File: products/prod-bdb1634806de/backend/app/schemas/advisory.py
from pydantic import BaseModel
from typing import Optional, List

class HazardResponse(BaseModel):
    type: str
    level: str
    measurement: Optional[str] = None
    distance_km: Optional[float] = None
    receipt: Optional[str] = None
    timestamp: Optional[str] = None
    is_cached: bool = False
    sim: bool = False

class ThresholdInfo(BaseModel):
    name: str
    condition: str
    fired: bool

class OverallResponse(BaseModel):
    level: str
    reason: str
    receipt: Optional[str] = None

class AdvisoryResponse(BaseModel):
    overall: OverallResponse
    hazards: List[HazardResponse]
    thresholds: List[ThresholdInfo]
    location: dict

File: products/prod-bdb1634806de/frontend/src/api/advisory.ts
import apiClient from './client';

export interface AdvisoryResponse {
  overall: {
    level: string;
    reason: string;
    receipt?: string;
  };
  hazards: Array<{
    type: string;
    level: string;
    measurement?: string;
    distance_km?: number;
    receipt?: string;
    timestamp?: string;
    is_cached: boolean;
    sim: boolean;
  }>;
  thresholds: Array<{
    name: string;
    condition: string;
    fired: boolean;
  }>;
  location: { lat: number; lon: number; rounded: boolean };
}

export async function fetchAdvisory(lat: number, lon: number): Promise<AdvisoryResponse> {
  const response = await apiClient.get<AdvisoryResponse>('/api/advisory', {
    params: { lat, lon },
  });
  return response.data;
}
